Private by default launch flow

niranjan-uma-shankar opened the pull request:
### Changes proposed in this Pull Request

* This PR implements the Private by default launch flow.
* The following happens when a user clicks the launch button on a private site:
    - If the site does not have a domain and/or a plan, we will take them through a `site-launch` flow. 
    This flow will show the domain step and the plan step before taking them to the preview page with a 
    successful launch banner:
    - If the site has both a domain and a plan, then the user is directly taken to the Preview page with a  launch banner celebrating the success.

<img width="1658" alt="screenshot 2019-01-29 at 5 25 39 pm" src="https://user-images.githubusercontent.com/1269602/51906793-fcac9900-23ea-11e9-8253-bfd2d7cf1a3a.png">

    
### Testing instructions

Please apply D23389-code and follow the testing instructions given there.

#### Scenario-1 - Launch a site that does not have either a domain or a plan

1. Create a site on the Free plan without a domain (except the default *.wordpress.com domain)
2. Clicking `Launch Site` should go through the launch flow, showing both the domain and plan steps.
There are a few scenarios that can be tested here:
    - Add both a domain and a plan to your cart so that you're taken to the checkout page. At this point, cancelling the purchase (by navigating away from checkout either by clicking the Trash icon in the cart on the left menu, or just closing the browser tab) should still launch the site. Check back in Preview that you no longer see the Launch Site button.
    - After adding both a domain and plan to your cart, make the purchase at the Checkout page. You should be redirected to the Site preview page with a success banner message.
    - You should be allowed to skip both the Domain purchase (by clicking Skip Purchase in the domain step) and Plan purchase (by clicking the Keep this Plan button in the plan step). You will then be directly taken to the Preview page with success banner (no checkout page)

#### Scenario-2 - Launch a site that does not have a domain but has a plan

1. Create a site with a paid plan but don't purchase a domain.
2. Clicking `Launch Site` should go through the launch flow, showing only the domain step. Test both with purchasing a domain and skipping Purchase - in both cases, it should take you to the Preview page with Success banner.

#### Scenario-3 - Launch a site that has a domain but does not have a plan
1. Create a site with a domain but no plan (for this you need to first create a site on a paid plan, purchase a domain, and then cancel the Plan purchase while retaining the domain).
2. Clicking `Launch Site` should go through the launch flow, showing only the plan step. Test both with purchasing a plan and skipping Purchase - in both cases, it should take you to the Preview page with Success banner.

#### Scenario-4 - Launch a site that has both a domain and a plan

1. Create a site on a paid plan with a domain
2. Launch the site, it should directly take you to the preview page (without going through the launch flow) and show a success banner.

#### Other tests

1. Test signup works correctly via the main and onboarding flows. 
2. Test Add a new site option works correctly.
3. Test the subdomain flow works correctly. This is being used in .blog landing pages (photo.blog, food.blog etc).  `/start/subdomain?vertical=a8c.8` should show the default recommendation on the domain step to be a *.food.blog.

niranjan-uma-shankar commented:
Ooops... I did something while rebase that closed this PR. Re-opening.

scruffian commented:
I made some changes to the cart which mean you no longer to supply the site Id - you can just use the site slug. That might simplify this code a bit...

scruffian commented on client/lib/signup/step-actions.js:
```diff
@@ -355,10 +317,85 @@ export function addPlanToCart( callback, { siteSlug }, { cartItem } ) {
+		addToCartAndProceed();
+	}
+}
+export function addDomainToCart(
```
I think we have several different ways of adding things to the cart. There's one for domains and one for plans. It would be good if they both used the same code.

niranjan-uma-shankar replied:
Please check [829ce5d](https://github.com/Automattic/wp-calypso/pull/30242/commits/829ce5d2df991c4e5cbaf0c2876dac789c126892). I've consolidated the common parts of the code in `processItemCart()`. I still had to retain `addPlanToCart()` and `addDomainToCart()` as separate functions because they provide different dependencies. Does it align with your suggestion or am I missing something?

scruffian commented on client/signup/steps/launch-site/index.jsx:
```diff
@@ -0,0 +1,52 @@
+		launchPrivateSite: () => {
+			const { flowName, stepName, stepSectionName, goToNextStep } = ownProps;
+
+			dispatch( launchSite( ownProps.signupDependencies.siteId ) );
```
Would it make more sense to make this the `apiRequestFunction` of the step, since that's what's happening behind the scenes?

niranjan-uma-shankar replied:
Agree. Implemented in [d21ca5a](https://github.com/Automattic/wp-calypso/pull/30242/commits/d21ca5a8f0eb32b50a2c80e50a730d494a1bfc21).

scruffian commented on client/signup/main.jsx:
```diff
@@ -309,6 +314,22 @@ class Signup extends React.Component {
 
+		if ( 'undefined' !== typeof excludePlanStep ) {
+			let planStepName = '';
+			if ( 'site-launch' === flowName ) {
```
It worries me when I see flow names in generic code. It looks like this is being implemented in a very specific way rather than a reusable way.

niranjan-uma-shankar replied:
I've removed specific references and made it more generic. Thanks for the tip.

a8ci18n commented on client/components/domains/domain-skip-suggestion/index.jsx:
```diff
@@ -0,0 +1,53 @@
+
+	render() {
+		const { translate } = this.props;
+		const buttonContent = translate( 'Skip Purchase', {
```
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/components/domains/domain-skip-suggestion/index.jsx:
```diff
@@ -0,0 +1,53 @@
+				<div className="domain-transfer-suggestion__domain-description">
+					<h3>{ this.props.selectedSiteSlug }</h3>
+					<p>
+						{ translate( 'This is your current free site address', {
```
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

a8ci18n commented on client/my-sites/plan-features/actions.jsx:
```diff
@@ -69,7 +70,11 @@ const PlanFeaturesActions = ( {
-		if ( isInSignup ) {
+		if ( isLaunchPage ) {
+			if ( freePlan ) {
+				buttonText = translate( 'Keep this plan', { context: 'button' } );
```
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

a8ci18n commented on client/signup/config/steps-pure.js:
```diff
@@ -148,6 +164,19 @@ export function generateSteps( {
+			providesDependencies: [ 'cartItem' ],
+			props: {
+				headerText: i18n.translate( 'Getting ready to launch your website' ),
+				subHeaderText: i18n.translate( "Pick a plan that's right for you" ),
```
Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'Pick a plan that's right for you.' )` *ES Score: 14*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/private-by-default-launch-flow)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

scruffian commented on client/lib/signup/step-actions.js:
```diff
@@ -346,17 +309,110 @@ export function getUsernameSuggestion( username, reduxState ) {
+	const siteId = getSiteId( reduxStore.getState(), siteSlug );
+	const providedDependencies = { siteSlug };
+
+	wpcom.undocumented().launchSite( siteId, function( error ) {
```
I think the API will also accept a siteSlug, so we can simplify this code.

niranjan-uma-shankar replied:
Implemented in [b2fcae0](https://github.com/Automattic/wp-calypso/pull/30242/commits/b2fcae034db8926a6854201af72a1966f6b32963).

niranjan-uma-shankar commented:
There are a few places where the destination is not respected. 
1. On the checkout page, if you click the Trash icon on the cart in the left menu, you will be taken to the Upgrades page
2. If you go through the launch flow on a site that has a plan but without a domain, then the checkout page will take you to `/thank-you/{siteSlug}/{receiptId}`
3. If you purchase a domain or plan in the launch flow, then you'll be taken to Calypso preview /view/{siteSlug}  instead of to https://{siteSlug}

At least some of them are due to the destination not being respected, and could get fixed with the work happening in https://github.com/Automattic/wp-calypso/pull/29136

fditrapani commented:
Thanks for all this work @niranjan-uma-shankar and @scruffian. It tested well with a free plan, paid plan plus subdomain, and paid plan plus a domain. Here are a couple notes:
- It would be good to get a video/animated gif walkthrough of the flow as part of the description of this PR. It took me some time to get this all running on my machine. 
- Clicking on the Launch button opens in a new tab feels disorienting to me. I think it would work better in the same tab. 

These items can most likely be addressed in separate PRs:
- Domains screen: I like the suggestions there but it's not super clear right off the bat that they are suggestions. 
- Domains: The skip option isn't super obvious because it's bundled in with the other choices. 
- Domains: If I search and don't like the options, it's not clear to me that I have to clear the search to see the skip option again. 
- Domains: The "Skip purchase" call to action doesn't seem right because we're asking folks to purchase a plan and not a domain.
- Plans: The "upgrade" call to action doesn't quite seem right — I do like the "Keep this plan" one though.

cc @danhauk

niranjan-uma-shankar commented:
The success launch banner(shown after launch) and the private site banner(before launch) both have the same icon. Maybe we could have different icons to show a difference.

Pre-address verification:
<img width="1396" alt="screenshot 2019-01-30 at 1 41 02 am" src="https://user-images.githubusercontent.com/1269602/51937283-44094880-2430-11e9-8060-8b834152cf5c.png">

Pre-launch:

<img width="1409" alt="screenshot 2019-01-30 at 1 41 36 am" src="https://user-images.githubusercontent.com/1269602/51937296-4a97c000-2430-11e9-8005-78ea23f55682.png">


Post-launch:

<img width="1658" alt="screenshot 2019-01-29 at 5 25 39 pm" src="https://user-images.githubusercontent.com/1269602/51937314-571c1880-2430-11e9-9c1a-01e0f98b5711.png">

danhauk commented:
> Maybe we could have different icons to show a difference.

I agree. It would be great to have something more celebratory there. I'll work on something for that.

adamziel commented:
> Domains: The "Skip purchase" call to action doesn't seem right because we're asking folks to purchase a plan and not a domain.

We definitely need to make it more clear. @fditrapani would you mock up some solution?

adamziel commented:
> Plans: The "upgrade" call to action doesn't quite seem right — I do like the "Keep this plan" one though.

How would you word it differently? "Buy"?

adamziel commented:
This is amazing overall, thank you for all the time you put in that! I reviewed the code - it looks good and works well! We can hammer out UX matters in a separate PR since private by default is disabled at the moment. Let's get e2e tests working, another pair of eyes on the review, and we could merge this one!

scruffian commented on client/signup/config/flows-pure.js:
```diff
@@ -310,6 +310,14 @@ export function generateFlows( { getSiteDestination = noop, getPostsDestination
 		lastModified: '2018-10-22',
 	};
 
+	flows[ 'site-launch' ] = {
```
Nitpick, but I'd call this `launch-site`

scruffian commented on client/signup/config/steps-pure.js:
```diff
@@ -510,6 +540,14 @@ export function generateSteps( {
+			apiRequestFunction: launchSiteApi,
+			dependencies: [ 'siteSlug' ],
+			providesDependencies: [ 'siteSlug' ],
+			delayApiRequestUntilComplete: true,
```
I don't really understand the need for `delayApiRequestUntilComplete`. What happens if you remove it from here?

niranjan-uma-shankar replied:
Agree, this can be removed. I've also removed it from the `domains-launch` step. Implemented in [b2fcae0](https://github.com/Automattic/wp-calypso/pull/30242/commits/b2fcae034db8926a6854201af72a1966f6b32963).

scruffian commented on client/signup/config/steps-pure.js:
```diff
@@ -510,6 +540,14 @@ export function generateSteps( {
+			stepName: 'launch',
+			apiRequestFunction: launchSiteApi,
+			dependencies: [ 'siteSlug' ],
+			providesDependencies: [ 'siteSlug' ],
```
Does this step need to provide a `siteSlug`, if its already being provided by another step in the flow?

niranjan-uma-shankar replied:
Correct, it doesn't need `siteSlug` to be provided here since it's already provided as a `providesDependenciesInQuery` in flows-pure.js. I have also removed `siteSlug` from the domains-launch step's `providesDependencies`.

Implemented in [b2fcae0](https://github.com/Automattic/wp-calypso/pull/30242/commits/b2fcae034db8926a6854201af72a1966f6b32963).

scruffian commented on client/signup/main.jsx:
```diff
@@ -256,7 +256,12 @@ class Signup extends React.Component {
+					vertical,
+					site_type: siteType,
+					exclude_plan: excludePlanStep,
+					exclude_domain: excludeDomainStep,
```
Are these coming from the query string? Would it make more sense to handle this at the framework level, and have mechanism in place in to skip steps in certain conditions? We could add some logic to the `domains-launch` and `plans-launch` steps which specifies the conditions under which they should be skipped.

niranjan-uma-shankar replied:
I've made a change in [2496746](https://github.com/Automattic/wp-calypso/pull/30242/commits/24967469a52b48ab9506c66038e1d111aaba5af0) to get the domains and plans for a site from the state tree itself. This works, but I'm not yet fully confident if the prop values `siteDomains` and `isPaidPlan` are always available in the component. I think it's in the right direction but needs a bit more testing.

niranjan-uma-shankar replied:
@scruffian Tested this for a while, and there is some inconsistent behaviour. 
On a free site with a registered domain, when I click the Launch Site button a few times, most of the times it correctly skips the domains step but there are a few times that it still shows the domain step. What I did:
1. Created a new private site with a plan + domain.
2. Cancelled the plan but kept the domain
3. Clicked the Launch Site button

The issue I see is that the `getState().sites.domains.items` is not always getting the domain values for the current site in the launch flow tab. It's somewhat random but reproducible. Would you be knowing why this is so?

scruffian replied:
I wonder if the API isn't always getting called. Maybe you could add the QuerySiteDomains component to ensure that it is?

niranjan-uma-shankar replied:
@scruffian  I've pushed in a fix that uses the QuerySiteDomains component, but the issue still persists 😕 

Also, I’m not sure if skipping steps in the framework would always work. For example, we want to skip domain step in the launch flow but no in the Add Site flow. In both flows, the state tree has a `siteId`value so we would then have to add specific conditions such as skip if in launch flow but not otherwise.

niranjan-uma-shankar replied:
@scruffian This is now fixed in [ad07829](https://github.com/Automattic/wp-calypso/pull/30242/commits/ad078297780801cc3c03c0078eb69abfadbc73d6). The skipping of steps is now handled at the framework level.

scruffian commented on client/signup/steps/domains/index.jsx:
```diff
@@ -180,6 +182,32 @@ class DomainsStep extends React.Component {
 	};
 
+	handleSkip = () => {
+		const domainItem = undefined,
```
It feels a bit messy to have to set all of these constants to undefined. Couldn't we modify the things that `submitSignupStep` expects so that it no longer requires all of these?

niranjan-uma-shankar replied:
Done in [b078ac7](https://github.com/Automattic/wp-calypso/pull/30242/commits/b078ac7b308a202f17954378a8846e7cb7d979ca).

danhauk commented:
I spent a little time yesterday working through some of the issues @fditrapani raised.

![domain search](https://user-images.githubusercontent.com/448298/51992473-0b429100-247b-11e9-8b3f-644f0022b8df.png)

For the domain search, the current domain should always be visible, as well as an option to skip choosing a new domain. We also want to make sure people know that if they skip it they can choose to do this at a later time. I think `Keep for now` indicates the skip functionality and be able to change it later, but open to other suggestions.

I also added a title over the default suggestions. This is something we may want to add for the regular add domain screen at `/domains/add/{:siteId}`. It makes it more clear these two that we show are automatic suggestions.

![select plan](https://user-images.githubusercontent.com/448298/51992651-63799300-247b-11e9-9e7a-21dfd0756a32.png)

I also really like `Keep this plan` for the plan you're currently on. For the others, rather than `Upgrade` we could try `Select [Plan name]`.

Thoughts, questions, concerns?

cc @adamziel @niranjan-uma-shankar @scruffian

scruffian commented:
@danhauk I like this. I feel concerned that users might miss the "keep for now" option on the domains.

scruffian commented on client/lib/wpcom-undocumented/lib/undocumented.js:
```diff
@@ -1492,6 +1492,18 @@ Undocumented.prototype.sitesNew = function( query, fn ) {
+/**
+ * Launches a private site
+ *
+ * @param {string} - ID of the site to be launched
```
Probably should clarify that this can also be a slug

niranjan-uma-shankar replied:
Good point. Implemented in [cc0060c](https://github.com/Automattic/wp-calypso/pull/30242/commits/cc0060c9055f4b32abdb90a0c063458932d9d960).

scruffian commented on client/signup/steps/domains/index.jsx:
```diff
@@ -180,6 +182,14 @@ class DomainsStep extends React.Component {
 		return `${ repo }/${ themeSlug }`;
 	};
 
+	handleSkip = () => {
```
This looks great now 👍

niranjan-uma-shankar commented on client/signup/main.jsx:
```diff
@@ -307,12 +322,46 @@ class Signup extends React.Component {
 		}
+	};
+
+	removeOtherFulfilledSteps = nextProps => {
```
Nice suggestion @scruffian, thanks! I've now pushed a refactored change that makes this function more abstract and processes all steps for fulfilled dependencies. Also, I could get rid of of `submitQueryDependencies()` which was specific to steps. Changes incorporated in [af0f360](https://github.com/Automattic/wp-calypso/pull/30242/commits/af0f36037d33d420a3555fc4aeab538440449ba7), [0dede0f](https://github.com/Automattic/wp-calypso/pull/30242/commits/0dede0f1d3773be4cafd0a0e17de3d8dd2a72d15) and [27d2edd](https://github.com/Automattic/wp-calypso/pull/30242/commits/27d2edde874f6764457853790ecba99897baf8a9).

scruffian replied:
This looks great. I'm loving your work!

fditrapani commented:
I took a little walk through the flow with Cain and it's looking good. Once we get this PR shipped, we can start working on the design changes @danhauk has been working on.

ramonjd commented:
Went through all scenarios that don't require purchasing a domain - time was against me. Worked as described. Great job.

One thing I noticed was that clicking **Remove** on the preview page only removes the banner _when you're on that page_. Returning from another page in the app, you see it again.

https://cldup.com/fo9-uvp_Jx.mp4

This is not a big deal, just wondering what **Remove** means in this context. What if I don't care about launching and wish to dismiss it permanently?

ramonjd commented on client/my-sites/plan-features/actions.jsx:
```diff
@@ -69,7 +70,14 @@ const PlanFeaturesActions = ( {
 		}
-		if ( isInSignup ) {
+		if ( isLaunchPage ) {
+			if ( freePlan ) {
```
Would it still work if we were to combine the conditions into one statement?

```
if ( isLaunchPage && freePlan ) {
```

niranjan-uma-shankar replied:
I've fixed this in [fa5d7ad](https://github.com/Automattic/wp-calypso/pull/30242/commits/fa5d7ad3e87b2d3da2cff9d839e1193845dcb35c) which adds the else part to the logic( button text as suggested by Dan [in this comment](https://github.com/Automattic/wp-calypso/pull/30242#issuecomment-458991265)).

([5a4966e](https://github.com/Automattic/wp-calypso/pull/30242/commits/5a4966e655218512c62b2e0fa69870e72897899e) fixes the typo in the above commit)

ramonjd commented on client/signup/steps/plans/index.jsx:
```diff
@@ -129,6 +129,7 @@ export class PlansStep extends Component {
 					site={ selectedSite || {} } // `PlanFeaturesMain` expects a default prop of `{}` if no site is provided
 					hideFreePlan={ hideFreePlan }
 					isInSignup={ true }
+					isLaunchPage={ this.props.isLaunchPage }
```
`this.props` is already "_destructured_". Not mandatory, but we could add it to the others, e.g.,

```
		const {
			disableBloggerPlanWithNonBlogDomain,
			hideFreePlan,
			isDomainOnly,
                         isLaunchPage,
			selectedSite,
		} = this.props;
```

niranjan-uma-shankar replied:
Agree, fixed in [0eb9cb6](https://github.com/Automattic/wp-calypso/pull/30242/commits/0eb9cb68eb1ab7fa50fdaa7a73353596c3f1f88f).

ramonjd commented on client/signup/main.jsx:
```diff
@@ -104,13 +109,18 @@ class Signup extends React.Component {
 		};
+
+		this.isPlanStepFulfilled = false;
+		this.isDomainStepFulfilled = false;
```
Are these two values used anywhere? I can't find any usages.

niranjan-uma-shankar replied:
You're right, removed them in [c220db7](https://github.com/Automattic/wp-calypso/pull/30242/commits/c220db77ebd6076fe966001d6e23153eb55f5004).

a8ci18n commented on client/my-sites/plan-features/actions.jsx:
```diff
@@ -69,7 +70,22 @@ const PlanFeaturesActions = ( {
+						'A selection to keep the current plan. Check screenshot - https://cloudup.com/cb_9FMG_R01',
+				} );
+			} else {
+				buttonText = translate( 'Select %(plan)s', {
```
Hi! I've found a possible matching string that has already been translated **13** times:
`translate( 'Select %(plan)s', { context: 'Button to select a paid plan by plan name, e.g., "Select Personal"'} )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

niranjan-uma-shankar commented:
Thanks for the review @ramonjd, pushed in a couple of changes that address your reviews.

> This is not a big deal, just wondering what **Remove** means in this context. What if I don't care about launching and wish to dismiss it permanently?

Good point. Dismissing it permanently seems like the right thing to do. I'm compiling a list of UX issues on this flow, and I will make sure to include your point. @fditrapani [suggested](https://github.com/Automattic/wp-calypso/pull/30242#issuecomment-458569571) to follow up with separate PRs for the UX fixes.

niranjan-uma-shankar commented:
The failing e2e test seems unrelated - when I run the tests locally, I don't see the error.

southp commented:
@niranjan-uma-shankar I'm not sure why but I can reproduce the same error locally by running `./node_modules/.bin/mocha ./specs/wp-signup-spec.js`. To match what happened in the CI result, you can also choose to run `env BROWSERSIZE=mobile ./node_modules/.bin/mocha ./specs/wp-signup-spec.js`. They don't make any difference on my end, though.

scruffian commented:
The tests pass! Great job @niranjan-uma-shankar

niranjan-uma-shankar commented:
Thanks @scruffian . 

I've fixed the failing tests by adding a site-topic step to the `subdomain` flow. All the .blog landing pages link to this flow with a vertical query parameter, so the site topic step should never be seen (since it'll get removed as a fulfilled step). However, accessing the `subdomain` step without the vertical query parameter will now show a new site topic step but it looks like this flow is never called directly without the query parameter.

michaeldcain commented:
@niranjan-uma-shankar: I just wanted to note that the domain mapping option in the launch flow didn't work for me or @fditrapani when we ran through this. It can probably be tracked in a separate PR, since this is close to merging and we'll be doing a round of updates before making it live.

southp commented:
I've bumped into a few issues along the way. The test plan is quite comprehensive and I'm not quite familiar with this, so there was a chance that I did something wrong though.

Here is a summary of what I saw.

### Scenario-1 - Launch a site that does not have either a domain or a plan

Works as expected.

### Scenario-2 - Launch a site that does not have a domain but has a plan

At the domain step, it didn't reflect the fact that the plan came with a free domain:
![demo-domain-step-only](https://user-images.githubusercontent.com/1842898/52702488-3b2f7f80-2fb7-11e9-8fcb-8a2970110076.gif)

Not sure if this is expected, though.

### Scenario-3 - Launch a site that has a domain but does not have a plan

Is our Blogger plan .blog only? I'm asking because it is selectable only when a .blog domain is chosen, but I'm not sure if that's a hard restriction. In this flow, I could first choose a non-blog domain with a plan, cancel the plan, and then choose the Blogger plan. If it is restricted to be bound with a .blog domain, we probably should handle that here.

### Scenario-4 - Launch a site that has both a domain and a plan


### Misc.

* The "Launch site" button in the site settings page doesn't follow the same flow at the moment so I could just launch the site without going through the site launch flow. Is it expected?
![demo-launch-from-setting](https://user-images.githubusercontent.com/1842898/52702881-17206e00-2fb8-11e9-9d2e-dbb85ef1b375.gif)
* On this branch, when I went through the signup flow with a plan, the processing page would appear twice. I'm not sure if this is also happening for others or on the master branch.

scruffian commented:
This PR is already big. I suggest we open issues for the remaining things here and get it merged.

niranjan-uma-shankar commented:
Thanks for the thorough testing @southp!

> At the domain step, it didn't reflect the fact that the plan came with a free domain

Thanks for reporting. I'm already tracking this separately in Trello as an issue to be fixed in a follow up PR. I think it's non-blocker for this PR but important to be fixed, and we anyway have a few updates that will go in before making private by default live to users(it is in a 0/100 ab test).

> Is our Blogger plan .blog only? I'm asking because it is selectable only when a .blog domain is chosen, but I'm not sure if that's a hard restriction. In this flow, I could first choose a non-blog domain with a plan, cancel the plan, and then choose the Blogger plan. If it is restricted to be bound with a .blog domain, we probably should handle that here.

The blogger plan allows only .blog domains. If you have a non-blog domain on a free site, then selecting a Blogger plan should auto-upgrade the cart to a Personal plan. #29899 and #30064 (and the backend patches referenced on those PRs) implemented a lot of the related work on this. 
So, the Blogger plan is presented but the Checkout page will upgrade the cart to a Personal plan(this is by design). Were you seeing the Blogger plan in the checkout page too with a non-blog domain? 

> When I launched the site from the site instead of the site view iframe, my site was launched successfully with a bit glitch

Good catch. Do you think we can address this in a follow up PR?

> The "Launch site" button in the site settings page doesn't follow the same flow at the moment so I could just launch the site without going through the site launch flow. Is it expected?

Adding to your point, another place this happens is in the Checklist - we have a Launch site option there too. I missed to transfer this point from my notes, so thanks for bringing it up(good catch, again). It does seem logical to have the launch flow linked in all places we allow the user to Launch, but I'm going to ping @scruffian @fditrapani and @michaeldcain for opinions. 

> On this branch, when I went through the signup flow with a plan, the processing page would appear twice. I'm not sure if this is also happening for others or on the master branch.

You're right. I strongly suspect this [line](https://github.com/Automattic/wp-calypso/pull/30242/files#diff-7b0ca7ad49dc313c6af0636bf22edd4eR571) which re-renders to fetch site domains. I'll take a closer look on how to avoid the the processing screen from showing twice.

fditrapani commented:
> This PR is already big. I suggest we open issues for the remaining things here and get it merged.

Good point by @scruffian 

> The "Launch site" button in the site settings page doesn't follow the same flow at the moment so I could just launch the site without going through the site launch flow. Is it expected?

We should be sending people to this flow where ever they might click Launch from.

southp commented:
> This PR is already big. I suggest we open issues for the remaining things here and get it merged.

Agreed 👍

niranjan-uma-shankar commented:
> Lets merge this and deal with the remaining issues in another PR. @niranjan-uma-shankar are you ok to create a new issue which compiles the remaining unaddressed feedback?

Happy to do it @scruffian.

scruffian commented:
Great 🚢
